optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolverJob.java

```diff
@@ -222,6 +245,14 @@ void close() {
         }
     }
 
+    void reset(Function<? super ProblemId_, ? extends Solution_> problemFinder) {
```

Comment:
I don't like the fact that I am resetting some initial state of the `SolverJob`. However, I haven't found any other way of keeping the `SolverJob` instance after the `reloadProblem()` call. If that is not a concern, I would prefer to kill the old `SolverJob` instance and create a new one.